netbox/dcim/forms/bulk_import.py

```diff
@@ -1161,27 +1161,35 @@ def __init__(self, *args, **kwargs):
         else:
             self.fields['parent'].queryset = InventoryItem.objects.none()
 
-    def clean_component_name(self):
-        content_type = self.cleaned_data.get('component_type')
-        component_name = self.cleaned_data.get('component_name')
+    def clean(self):
+        cleaned_data = self.cleaned_data
+        content_type = cleaned_data.get('component_type')
+        component_name = cleaned_data.get('component_name')
         device = self.cleaned_data.get("device")
 
-        if not device and hasattr(self, 'instance') and hasattr(self.instance, 'device'):
-            device = self.instance.device
-
-        if not all([device, content_type, component_name]):
-            return None
-
-        model = content_type.model_class()
-        try:
-            component = model.objects.get(device=device, name=component_name)
-            self.instance.component = component
-        except ObjectDoesNotExist:
-            raise forms.ValidationError(
-                _("Component not found: {device} - {component_name}").format(
-                    device=device, component_name=component_name
+        if content_type:
+            if device is None:
+                cleaned_data.pop('component_type')
```

Comment:
We generally want to avoid mutating the cleaned data. If `device` (a required field) hasn't been specified, we should be able to abort here as it will have already raised a validation error.